libnavigation-core/src/main/java/com/mapbox/navigation/core/navigator/NavigatorMapper.kt

```diff
@@ -10,6 +15,26 @@ internal fun TripStatus.getMapMatcherResult(): MapMatcherResult {
         navigationStatus.offRoadProba > 0.5,
         navigationStatus.offRoadProba,
         navigationStatus.map_matcher_output.isTeleport,
+        navigationStatus.prepareSpeedLimit(),
         navigationStatus.map_matcher_output.matches.firstOrNull()?.proba ?: 0f
     )
 }
+
+internal fun NavigationStatus.prepareSpeedLimit(): SpeedLimit? {
```

Comment:
Tests are failing now because of this change 👀 

```
> Task :libnavigation-core:compileDebugUnitTestKotlin FAILED
e: /root/code/libnavigation-core/src/test/java/com/******/navigation/core/navigator/NavigatorMapperTest.kt: (31, 46): Type mismatch: inferred type is com.******.navigation.base.speed.model.SpeedLimit? but com.******.navigator.SpeedLimit? was expected
e: /root/code/libnavigation-core/src/test/java/com/******/navigation/core/navigator/NavigatorMapperTest.kt: (70, 46): Type mismatch: inferred type is com.******.navigation.base.speed.model.SpeedLimit? but com.******.navigator.SpeedLimit? was expected
e: /root/code/libnavigation-core/src/test/java/com/******/navigation/core/navigator/NavigatorMapperTest.kt: (109, 46): Type mismatch: inferred type is com.******.navigation.base.speed.model.SpeedLimit? but com.******.navigator.SpeedLimit? was expected
e: /root/code/libnavigation-core/src/test/java/com/******/navigation/core/navigator/NavigatorMapperTest.kt: (148, 46): Type mismatch: inferred type is com.******.navigation.base.speed.model.SpeedLimit? but com.******.navigator.SpeedLimit? was expected
e: /root/code/libnavigation-core/src/test/java/com/******/navigation/core/navigator/NavigatorMapperTest.kt: (187, 46): Type mismatch: inferred type is com.******.navigation.base.speed.model.SpeedLimit? but com.******.navigator.SpeedLimit? was expected

> Task :libnavigation-metrics:bundleLibCompileToJarRelease
> Task :libnavigation-core:parseReleaseLocalResources
> Task :libnavigation-router:bundleLibRuntimeToJarDebug
> Task :libnavigation-core:processReleaseManifest

FAILURE: Build failed with an exception.
```

Refs. https://github.com/mapbox/mapbox-navigation-android/pull/3839#discussion_r532973137